pkg/counter/counter.go

```diff
@@ -1,96 +1,66 @@
 package counter
 
 import (
+	"errors"
 	"fmt"
+	"math"
 	"strconv"
 	"sync/atomic"
-
-	"github.com/aquasecurity/tracee/pkg/errfmt"
 )
 
 type Counter struct {
-	count uint64
+	value uint64
 }
 
-func NewCounter(v uint64) Counter {
-	return Counter{
-		count: v,
-	}
+// NewCounter creates a new counter with the given initial value.
+func NewCounter(initialValue uint64) Counter {
+	return Counter{value: initialValue}
 }
 
-func (c *Counter) Increment(amount ...uint64) error {
-	if len(amount) == 0 {
-		return c.incAtomic(1)
-	}
-
-	sum, err := sumUint64(amount...)
-	if err != nil {
-		return errfmt.WrapError(err)
+// Increase increases the counter by the given value (thread-safe).
+func (c *Counter) Increase(values ...uint64) error {
+	if len(values) == 0 {
+		values = append(values, 1)
 	}
 
-	return c.incAtomic(sum)
-}
-
-func (c *Counter) Decrement(amount ...uint64) error {
-	if len(amount) == 0 {
-		return c.decAtomic(1)
-	}
-
-	sum, err := sumUint64(amount...)
-	if err != nil {
-		return errfmt.WrapError(err)
+	for _, value := range values {
+		if atomic.AddUint64(&c.value, value) < value {
```

Comment:
So now the overflow will be detected in the middle of increasing and not ahead of time. Which means that once an error is returned the counter won't stay in its initial count, meaning there could be a "partial increase".

Not sure if this behavior is better or not, but pointing it out.

Reply:
Yes, I'm aware. The only way to guarantee multiple increases is to holding a mutex/lock, which is something we don't want I suppose.

Reply:
Ok, I understand the choice now.
I wonder if we should even allow the API of allowing more then one value increase at a time (I don't think we use it except for the shorthand of `.Increase() => increase 1`).